Dockerfile

```
@@ -0,0 +1,32 @@
+# Use the offical Python base image
+FROM python:3.11-slim
+
+# add curl as a very useful debugging tool not available in `slim`
+RUN apt-get update && apt-get install -y curl && rm -rf /var/lib/apt/lists/*
+
+# Set enviornment variables
+ENV PYTHONDONTWRITEBYTECODE=1
+
+# Set working dir in container
+WORKDIR /opt/app
+
+# Install build dependencies for projects using pyproject.toml
+RUN apt-get update && apt-get install -y --no-install-recommends \
+    build-essential \
+    && apt-get clean \
+    && rm -rf /var/lib/apt/lists/*
+
+# Copy pyproject.toml and requirements.txt first for dependency installation
+COPY pyproject.toml requirements.txt ./
+
+# Install dependencies
+RUN pip install --no-cache-dir -r requirements.txt
+
+# Copy the rest of the application code into the container
+COPY . .
+
+# Define the default command to run the application
+#ENTRYPOINT ["python", "timewarp/api.py"]
+#CMD ["main"]
+# Debugging endpoint - idle spin
+CMD ["tail", "-f", "/dev/null"]
```

Some leftover debugging here I think?

---

looking for advice on best invocation.  the commented out ENTRYPOINT would run `timewarp/api.py` and exit the container.  switched to the idle spin for interaction needed to copy files and then invoke with desired parameters.

suggestions @ivan-c ?

---

>looking for advice on best invocation. the commented out ENTRYPOINT would run timewarp/api.py and exit the container

that sounds correct. with entrypoint specified, you'd run it as follows:
`docker run $IMAGE /tmp http://fhir-internal:8080/fhir`

if you want, you can additionally use CMD to specify default args. if CMD is set (as in the suggestion), the args in CMD will be used by default, unless the user specifies their own eg:
`docker run $IMAGE /different-than-temp http://foo.bar:8080/fhir`


```suggestion
ENTRYPOINT ["python", "timewarp/api.py"]
CMD ["/tmp", "http://fhir-internal:8080/fhir"]
```